### DISCUSSION TITLE
Tasmota working without wifi connection

### DISCUSSION THREAD

**Discussion by gbananachewicz** [arendst/Tasmota#16495] [Q&A] [answered]
How can I set tasmota to work without wifi connection ?
Working as standalone device with logic fro berry.
It could make a wifi SSID and password to connect to it for settings. (Or if no possibility I will secure it with admin password)
There is no Ntp or something else needed. 
Just work without wifi and a possibility to programming from phone.

**Comment by barbudor** [comment, marked as answer]
That's not Tasmota use-case
WiFi and NTP are required although NTP could be substituted by GPS or Ds3231 clock chip

**Reply by gbananachewicz** [reply]
It can be 1900 year that's no problem for me or the esp board. Time is no needed for anything.

I want work device without wifi.

**Reply by barbudor** [reply]
It might not be a problem for you but it for Tasmota that hasn't been designed to work in those conditions
Without wifi and without proper date source Tasmota won't work properly
Sorry

**Reply by gbananachewicz** [reply]
Not designed . One more option needed. 
So762 work without wifi.

**Reply by sfromis** [reply]
This is far outside of what Tasmota was designed for. The "M" in the name literally means "MQTT", as it is meant to be used in conjunction with a smarthome backend dashboard/automation.

**Reply by gbananachewicz** [reply]
Are you closed for new inovatrions/ideas ??

Tasmota Relay with push button worka after restart without wifi. 
Autoexec.be  seems not load without wifi, after wifi connection all aitoexec.be commands working. 
Could someone fix it ?

**Reply by sfromis** [reply]
No, just explaining a bit about the project background.

If you have issues unrelated to your suggestion, this thread may not be a good place to get attention. But do make sure that details are what you think; I just did a test restarting a Tasmota device without Wifi enabled, and `autoexec.be` was executed just fine.

**Reply by gbananachewicz** [reply]
What do you have in autoexec ? Could you show it ? Then i would test it.

**Reply by gbananachewicz** [reply]
Make this working autoexec.be  without wifi
```
tasmota.add_rule('IBEACON#UID==A2FA7357C8CD4B9598FD9D091C43443',
  def (value,trigger,jsonmap)
    var rssi=jsonmap['IBEACON']['RSSI']
    if (rssi <= -10 && rssi >=-40)
      tasmota.set_power(0, true)
end
    if (rssi <= -41 && rssi >=-100)
      tasmota.set_power(0, false)
  print('speł 2 daleko')  
    end
if rssi >=0
      tasmota.set_power(0, false)
  print('speł 3  zginąl sygnał')  
    end

  end)
```

PS. Mayby Rule1 do not load ????     Can I make rules in autoexec.bat ??

**Reply by sfromis** [reply]
Example of `autoexec.be` tested to be working while having deactivated the configured Wifi:
```
tasmota.add_rule("ESP32#Temperature",/value,trigger,jsonmap->print(value,trigger,jsonmap))
print('trigger added')
```
Both the immediate print and the temperature messages appeared as expected. And of course also the very specific message from Tasmota (with `Weblog 3`):
```
00:00:00.579 trigger added
00:00:00.580 BRY: Successfully loaded 'autoexec.be'
00:00:06.700 QPC: Reset
00:00:08.057 39.36 ESP32#Temperature {'ESP32': {'Temperature': 39.36}}
```
Of course, to get the above I had a serial connection to the Tasmota device while booting.

What you posted indicates nothing about whether `autoexec.be` was executed or not. It merely suggests that your triggers were not firing the expected way, without you having done any debugging about why.

Anyway, this does not change the fact that no network connection (Wifi or Ethernet) is not a use case Tasmota is aiming to support.

If would not be particularly interesting creating classic rules in autoexec.bat, as those are persistent and available after boot without anything in autoexec.bat.

**Reply by gbananachewicz** [reply]
Why my autoexec code won't work without wifi ( code starting to work after connection to wifi or AP without internet). 
Turn led ON after boot work and autoexec seems to be loaded but the main functions of code do not working.
Please help.

**Reply by sfromis** [reply]
Better do some debugging, like what I told you about. Inserting print statements, and having serial connected is a good starting point, to work out what happens.

**Comment by gbananachewicz** [comment]
```
rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DOUT, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12616
load:0x40080400,len:2916
entry 0x400805c4

00:00:00.003-168/37 HDW: ESP32-D0WDQ6
00:00:00.041-166/36 UFS: FlashFS mounted with 296 kB free
00:00:00.078 CFG: Loaded from File, Count 95
00:00:00.094 QPC: Count 1
00:00:00.233 BRY: Berry initialized, RAM used=3735 bytes
00:00:00.257 Project tasmota - Tasmota Version 12.1.1(bluetooth)-2_0_4_1(2022-08-25T11:34:24)
00:00:00.259 iBeacon register for advert callbacks
00:00:00.260 MI32Option0: 1
00:00:00.271 MI32Option1: 0
00:00:00.272 MI32Option2: 0
00:00:00.272 MI32Option4: 1
00:00:00.273 MI32Option5: 0
00:00:00.274 MI32Option6: 0
00:00:00.285 M32: init: request callbacks
00:00:00.532 RSL: RESULT = {"POWER":"OFF"}
00:00:00.534 RSL: POWER = OFF
00:00:00.536 Hello, Grzes!
00:00:00.537 Stan blokady =  0
00:00:00.540 BRY: Successfully loaded 'autoexec.be'
00:00:01.248 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:04.261 WIF: Connect failed as AP cannot be reached
00:00:04.625 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:06.882 QPC: Reset
00:00:07.532 WIF: Connect failed as AP cannot be reached
00:00:08.884 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:12.851 WIF: Connect failed as AP cannot be reached
00:00:13.214 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:16.063 WIF: Connect failed as AP cannot be reached
00:00:17.427 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:20.324 WIF: Connect failed as AP cannot be reached
00:00:20.688 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:23.596 WIF: Connect failed as AP cannot be reached
00:00:24.949 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:27.802 WIF: Connect failed as AP cannot be reached
00:00:28.166 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:31.068 WIF: Connect failed as AP cannot be reached
00:00:32.434 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:35.334 WIF: Connect failed as AP cannot be reached
00:00:35.699 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:38.599 WIF: Connect failed as AP cannot be reached
00:00:39.964 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:42.818 WIF: Connect failed as AP cannot be reached
00:00:43.183 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:45.094 WIF: Connect failed as AP cannot be reached
00:00:46.363 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:47.893 WIF: Connected
00:00:48.156 HTP: Web server active on tasmota-2FC20C-0524 with IP address 192.168.43.234
00:00:48.497 BLE: MasterEnable->1
00:00:48.501 BLE: BLETask:Starting NimBLE
00:00:48.131 RSL: INFO1 = {"Info1":{"Module":"ESP32-DevKit","Version":"12.1.1(bluetooth)","FallbackTopic":"cmnd/DVES_2FC20C_fb/","GroupTopic":"cmnd/tasmotas/"}}
00:00:48.161 RSL: INFO2 = {"Info2":{"WebServerMode":"Admin","Hostname":"tasmota-2FC20C-0524","IPAddress":"192.168.43.234"}}
00:00:49.001 RSL: INFO3 = {"Info3":{"RestartReason":"Vbat power on reset","BootCount":45}}
00:00:49.018 RSL: RESULT = {"POWER":"OFF"}
00:00:49.023 RSL: POWER = OFF
00:00:49.080 RUL: SYSTEM#BOOT performs "backlog iBeacon 1; BLEAddrFilter 3; BlePeriod 1; iBeaconOnlyAliased 0; iBeaconPeriod 1; iBeaconTimeout 8; BLEMaxAge 4; BLEAlias

00:00:49.149 RSL: RESULT = {"IBEACON":1}
00:00:49.400 RSL: RESULT = {"BLEAddrFilter1":3}
00:00:49.490 RSL: SENSOR = {"Time":"1970-01-01T00:00:49","IBEACON":{"MAC":"0E2B7B70309A","RSSI":-81,"STATE":"ON","PERSEC":0}}
00:00:49.650 RSL: RESULT = {"BLEPeriod":"Done"}
00:00:50.081 RSL: RESULT = {"IBEACONonlyaliased":0}
00:00:50.220 RSL: SENSOR = {"Time":"1970-01-01T00:00:50","IBEACON":{"MAC":"C921E6EF8689","NAME":"GR_SW010","RSSI":-66,"STATE":"ON","PERSEC":0}}
00:00:50.256 RSL: RESULT = {"POWER":"ON"}
00:00:50.259 RSL: POWER = ON
00:00:50.264 1
00:00:50.269 spel 1 jest blisko
00:00:50.272  Blokada załączona na powtórne zadziałanie Zamka, blokada =  1
00:00:50.292 RSL: RESULT = {"IBEACONperiod":1}
00:00:50.548 RSL: RESULT = {"IBEACONtimeout":8}
00:00:50.873 RSL: SENSOR = {"Time":"1970-01-01T00:00:50","IBEACON":{"MAC":"0E2B7B70309A","RSSI":-74,"STATE":"ON","PERSEC":1}}
00:00:50.906 RSL: RESULT = {"BLEMaxAge1":4}
```


Afer connected Wifi starting  BLE: MasterEnable->1 and BLE: BLETask:Starting Nimble and it start working!

This is the Problem 

00:00:46.363 WIF: Connecting to AP1 Konekt in mode 11n as tasmota-2FC20C-0524...
00:00:47.893 WIF: Connected
00:00:48.156 HTP: Web server active on tasmota-2FC20C-0524 with IP address 192.168.43.234
00:00:48.497 BLE: MasterEnable->1
00:00:48.501 BLE: BLETask:Starting NimBLE

**Reply by sfromis** [reply]
So `autoexec.be` is running just fine, but the BLE support is meant for normal Tasmota operation, meaning Wifi available. Running without network is just way outside of how Tasmota is designed to be used. Thus, Tasmota seems to work as designed.

**Reply by gbananachewicz** [reply]
Could you tell me what to do with own compile to make bluetooth  working ?
I really need it to work without wifi.
